### PR TITLE
Z-Wave JS: Fix validation/parsing for custom config param UI

### PR DESCRIPTION
**Pull request by AlCalzone**
## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes the validation part of https://github.com/zwave-js/certification-backlog/issues/31#issuecomment-2470437479

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/integrations/integration-panels/zwave_js/zwave_js-custom-param.ts [hunk 1]
```diff
@@ -117,21 +117,32 @@ class ZWaveJSCustomParam extends LitElement {
     `;
   }
 
+  private _tryParseNumber(value: string): number | undefined {
+    if (!value) return;
+    const parsed = Number(value);
+    if (Number.isNaN(parsed)) return;
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zwave_js/zwave_js-custom-param.ts:123]:
```suggestion
    if (Number.isNaN(parsed)) return undefined;
```

File: src/panels/config/integrations/integration-panels/zwave_js/zwave_js-custom-param.ts [hunk 2]
```diff
@@ -117,21 +117,32 @@ class ZWaveJSCustomParam extends LitElement {
...
+    // eslint-disable-next-line consistent-return -- not sure why eslint expects no return here
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zwave_js/zwave_js-custom-param.ts:124]:
```suggestion
```

File: src/panels/config/integrations/integration-panels/zwave_js/zwave_js-custom-param.ts [hunk 3]
```diff
@@ -117,21 +117,32 @@ class ZWaveJSCustomParam extends LitElement {
     `;
   }
 
+  private _tryParseNumber(value: string): number | undefined {
+    if (!value) return;
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zwave_js/zwave_js-custom-param.ts:121]:
```suggestion
    if (!value) return undefined;
```

**Review by MindFreeze — APPROVED** [review]

File: src/panels/config/integrations/integration-panels/zwave_js/zwave_js-custom-param.ts (hunk 2 above)
Comment (AlCalzone) [reply src/panels/config/integrations/integration-panels/zwave_js/zwave_js-custom-param.ts]: Odd - was this not an error on CI? Only locally on my end?

**Pull request merged**